File: r/difference-of-squares/difference-of-squares.R
# Name: difference-of-squares.R
# Purpose: exercism, R track, difference-of-squares
#     this function takes a natural_number
#     and returns the difference-of-squares (\Sigma_1^n x)^2 - \Sigma_1^n x^2
# Written by Z Knight, 2019.10.07
#
difference_of_squares <- function(natural_number) {
  sum_squared <- (natural_number*(natural_number+1)/2)^2
  sum_of_squares <- sum(c(0:natural_number)^2)
  difference_of_squares <- sum_squared - sum_of_squares 
}

File: python/difference-of-squares/difference_of_squares.py
#! /usr/bin/env python
"""
    Name:
        difference_of_squares.py
    Purpose:
        exercism, python track, difference of squares: 
            Find the difference between the square of the sum and the sum of the squares of the first N natural numbers.
    Written by:
        Z Knight, 2020.09.12
"""
def square_of_sum(number):
    """
    
    """
    sum = (number*(number+1))/2
    return sum**2


def sum_of_squares(number):
    """

    """
    sum = 0
    for num in range(1, number+1):
        sum += num**2
    return sum


def difference_of_squares(number):
    """
    Purpose: 
        Find the difference between the square of the sum and the sum of the squares of the first N natural numbers.
    Inputs:
        Number: the limit N
    Returns:
        The difference
    """
    return square_of_sum(number)-sum_of_squares(number)

